gtsam/geometry/Pose2.cpp

```diff
@@ -309,54 +309,77 @@ double Pose2::range(const Pose2& pose,
- * using dp=[dpx;dpy] and q=[dqx;dqy] we have
- *  |dqx|   |c  -s|     |dpx|     |dpx -dpy|     |c|
+ * using db=[dbx;dby] and a=[dax;day] we have
+ *  |dax|   |c  -s|     |dbx|     |dbx -dby|     |c|
  *  |   | = |     |  *  |   |  =  |        |  *  | | = H_i*cs
- *  |dqy|   |s   c|     |dpy|     |dpy  dpx|     |s|
+ *  |day|   |s   c|     |dby|     |dby  dbx|     |s|
  * where the Hi are the 2*2 matrices. Then we will minimize the criterion
- * J = \sum_i norm(q_i - H_i * cs)
+ * J = \sum_i norm(a_i - H_i * cs)
  * Taking the derivative with respect to cs and setting to zero we have
- * cs = (\sum_i H_i' * q_i)/(\sum H_i'*H_i)
+ * cs = (\sum_i H_i' * a_i)/(\sum H_i'*H_i)
  * The hessian is diagonal and just divides by a constant, but this
  * normalization constant is irrelevant, since we take atan2.
- * i.e., cos ~ sum(dpx*dqx + dpy*dqy) and sin ~ sum(-dpy*dqx + dpx*dqy)
+ * i.e., cos ~ sum(dbx*dax + dby*day) and sin ~ sum(-dby*dax + dbx*day)
  * The translation is then found from the centroids
- * as they also satisfy cq = t + R*cp, hence t = cq - R*cp
+ * as they also satisfy ca = t + R*cb, hence t = ca - R*cb
  */
 
-boost::optional<Pose2> align(const vector<Point2Pair>& pairs) {
-
-  size_t n = pairs.size();
-  if (n<2) return boost::none; // we need at least two pairs
+boost::optional<Pose2> Pose2::Align(const Point2Pairs &ab_pairs) {
+  const size_t n = ab_pairs.size();
+  if (n < 2) {
+    return boost::none;  // we need at least 2 pairs
+  }
 
   // calculate centroids
-  Point2 cp(0,0), cq(0,0);
-  for(const Point2Pair& pair: pairs) {
-    cp += pair.first;
-    cq += pair.second;
+  Point2 ca(0, 0), cb(0, 0);
+  for (const Point2Pair& pair : ab_pairs) {
+    ca += pair.first;
+    cb += pair.second;
   }
-  double f = 1.0/n;
-  cp *= f; cq *= f;
+  const double f = 1.0/n;
+  ca *= f;
+  cb *= f;
 
   // calculate cos and sin
-  double c=0,s=0;
-  for(const Point2Pair& pair: pairs) {
-    Point2 dp = pair.first - cp;
-    Point2 dq = pair.second - cq;
-    c += dp.x() * dq.x() + dp.y() * dq.y();
-    s += -dp.y() * dq.x() + dp.x() * dq.y();
+  double c = 0, s = 0;
+  for (const Point2Pair& pair : ab_pairs) {
+    Point2 da = pair.first - ca;
+    Point2 db = pair.second - cb;
+    c += db.x() * da.x() + db.y() * da.y();
+    s += -db.y() * da.x() + db.x() * da.y();
   }
 
   // calculate angle and translation
-  double theta = atan2(s,c);
-  Rot2 R = Rot2::fromAngle(theta);
-  Point2 t = cq - R*cp;
+  const double theta = atan2(s, c);
+  const Rot2 R = Rot2::fromAngle(theta);
+  const Point2 t = ca - R*cb;
   return Pose2(R, t);
 }
 
+boost::optional<Pose2> Pose2::Align(const Matrix& a, const Matrix& b) {
+  if (a.rows() != 2 || b.rows() != 2 || a.cols() != b.cols()) {
+    throw std::invalid_argument(
+      "Pose2:Align expects 2*N matrices of equal shape.");
+  }
+  Point2Pairs ab_pairs;
+  for (size_t j=0; j < a.cols(); j++) {
+    ab_pairs.emplace_back(a.col(j), b.col(j));
+  }
+  return Pose2::Align(ab_pairs);
```

Comment:
I would go in the other direction, but okay.

Reply:
`aTb corresponds to (a,b) pairs`

Reply:
Oh, I agree with that! Other PR. Will merge now.